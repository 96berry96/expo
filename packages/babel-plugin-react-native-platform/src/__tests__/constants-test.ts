import { testCompressed, transform } from './utils';

it(`removes __DEV__ and process.env.NODE_ENV`, () => {
  const transpiledCode = transform(`
  if (__DEV__) {
    console.log("DEV MODE")
  } else {
    console.log("PROD MODE")
  }
  if (process.env.NODE_ENV === 'development') {
    console.log("DEV MODE")
  } else {
    console.log("PROD MODE")
  }
  `, {
    platform: `ios`,
    mode: 'development',
  });
  expect(transpiledCode).toMatchSnapshot();
  const minifiedCode = testCompressed(transpiledCode)
  expect(minifiedCode).toMatch('DEV');
  expect(minifiedCode).not.toMatch('PROD');
});

// react-native-web redefines this value.
// Terser should remove it during the bundling.
it(`keeps __DEV__ redefinition`, () => {
  const transpiledCode = transform(
    `
  const __DEV__ = process.env.NODE_ENV !== 'production';
  `,
    {
      platform: `web`,
      mode: 'development',
    }
  );

  expect(transpiledCode).toMatch('__DEV__');
  expect(transpiledCode).not.toMatch('process.env.NODE_ENV');
  testCompressed(transpiledCode)
});

// react-native-web redefines this value.
// Terser should remove it during the bundling.
it(`should pass over process.env.NODE_ENV redefinition`, () => {
  const transpiledCode = transform(
    `
  process.env.NODE_ENV = 'production';
  `,
    {
      platform: `web`,
      mode: 'development',
    }
  );
  expect(transpiledCode).toMatch(`process.env.NODE_ENV = 'production'`);
  testCompressed(transpiledCode)
});
