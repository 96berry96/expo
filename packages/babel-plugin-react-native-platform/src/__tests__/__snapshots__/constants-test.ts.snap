// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keeps __DEV__ redefinition 1`] = `"const __DEV__ = !0;"`;

exports[`removes __DEV__ and process.env.NODE_ENV 1`] = `
"{
  console.log(\\"DEV MODE\\");
}

if (\\"development\\" === 'development') {
  console.log(\\"DEV MODE\\");
} else {
  console.log(\\"PROD MODE\\");
}"
`;

exports[`removes __DEV__ and process.env.NODE_ENV 2`] = `"console.log(\\"DEV MODE\\"), console.log(\\"DEV MODE\\");"`;

exports[`should pass over process.env.NODE_ENV redefinition 1`] = `"process.env.NODE_ENV = \\"production\\";"`;
